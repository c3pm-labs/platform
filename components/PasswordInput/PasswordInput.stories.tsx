import { Formik, Form } from 'formik';
import { Meta } from '@storybook/react';
import { actions } from '@storybook/addon-actions';
import * as Yup from 'yup';

import PasswordInput, { PasswordInputProps } from '.';

export default {
  title: 'PasswordInput',
  argTypes: {
    label: {
      defaultValue: 'password',
      control: { type: 'text' },
    },
    placeholder: {
      defaultValue: '',
      control: { type: 'text' },
    },
    fullWidth: {
      defaultValue: false,
      control: { type: 'boolean' },
    },
    required: {
      defaultValue: false,
      control: { type: 'boolean' },
    },
  },
} as Meta;

const onSubmit = actions('onSubmit');

const passwordSchema = Yup.object().shape({
  password: Yup.string()
    .required('No password provided.')
    .min(8, 'Password is too short.'),
});

export const Basic = (args: PasswordInputProps): JSX.Element => (
  <Formik
    validationSchema={passwordSchema}
    initialValues={{ password: '' }}
    {...onSubmit}
  >
    <Form>
      <PasswordInput
        name="password"
        {...args}
      />
    </Form>
  </Formik>
);
