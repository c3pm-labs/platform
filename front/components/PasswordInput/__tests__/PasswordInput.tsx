import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import * as Yup from 'yup';
import React from 'react';
import { Formik, Form } from 'formik';

import PasswordInput from '../index';

test('PasswordInput unseen', () => {
  const passwordSchema = Yup.object().shape({
    password: Yup.string()
      .required('No password provided.')
      .min(8, 'Password is too short.'),
  });

  const { container } = render(
    <Formik
      validationSchema={passwordSchema}
      initialValues={{ password: '' }}
      // eslint-disable-next-line no-console
      onSubmit={(): void => console.log('submited')}
    >
      <Form>
        <PasswordInput name="password" label="toto" />
      </Form>
    </Formik>,
  );

  expect(container).toMatchSnapshot();
});


test('PasswordInput visible', () => {
  const passwordSchema = Yup.object().shape({
    password: Yup.string()
      .required('No password provided.')
      .min(8, 'Password is too short.'),
  });

  const { container } = render(
    <Formik
      validationSchema={passwordSchema}
      initialValues={{ password: '' }}
      // eslint-disable-next-line no-console
      onSubmit={(): void => console.log('submited')}
    >
      <Form>
        <PasswordInput name="password" label="toto" />
      </Form>
    </Formik>,
  );

  userEvent.click(screen.getByTestId('iconPasswordVisibility'))
  expect(container).toMatchSnapshot();
});
