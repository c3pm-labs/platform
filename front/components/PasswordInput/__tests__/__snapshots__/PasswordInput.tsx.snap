// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput unseen 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
        for="password"
        id="password-label"
      >
        toto
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-describedby="password-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
        >
          <button
            aria-label="toggle password visibility"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-root-1 MuiIconButton-edgeEnd"
            data-testid="iconPasswordVisibility"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-5"
          >
            <span>
              toto
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-root-2 MuiFormHelperText-contained MuiFormHelperText-marginDense"
        id="password-helper-text"
      >
        <span>
          ​
        </span>
      </p>
    </div>
  </form>
</div>
`;

exports[`PasswordInput visible 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
        for="password"
        id="password-label"
      >
        toto
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-describedby="password-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="password"
          name="password"
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
        >
          <button
            aria-label="toggle password visibility"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-root-7 MuiIconButton-edgeEnd"
            data-testid="iconPasswordVisibility"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            >
              <span
                class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
              >
                <span
                  class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
                />
              </span>
            </span>
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-11"
          >
            <span>
              toto
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-root-8 MuiFormHelperText-contained MuiFormHelperText-marginDense"
        id="password-helper-text"
      >
        <span>
          ​
        </span>
      </p>
    </div>
  </form>
</div>
`;
